hca/util/fs_helper.py

```diff
@@ -0,0 +1,80 @@
+import os, tempfile, errno
+from datetime import datetime
+
+
+class FSHelper(object):
+    """
+    Static helper class providing convenience methods for reading and writing files.
+    """
+
+    @staticmethod
+    def atomic_write(filename, shared_dir, content):
+        """
+        Atomically writes content a to file.
+
+        Writes to temp file and renames file (atomic operation) to desired filename.
+        To ensure atomicity, the src and dest dirs must be the same.
+
+        :param filename: File name only (e.g. example.txt)
+        :param shared_dir: Shared directory of temp and dest file
+        :param content: Content to write to file
+        """
+        dest_filename = os.path.join(shared_dir, filename)
+        try:
+            with tempfile.NamedTemporaryFile('wb', dir=shared_dir, delete=False) as temp:
+                temp_filename = temp.name
+                temp.write(content)
+            FSHelper._atomic_rename(temp_filename, dest_filename)
+        finally:
+            if temp_filename and os.path.isfile(temp_filename):
+                os.remove(temp_filename)
+
+    @staticmethod
+    def _atomic_rename(src, dest):
```

Comment:
suggestion:

`def _atomic_rename(src, dest, max_retries=5):`

Reply:
I like it